app/components/buttons/regular_button_component/regular_button_component.html.haml

```diff
@@ -0,0 +1,36 @@
+- slim_class = @size == "slim" ? "slim " : " "
+- danger_class = @color == "danger" ? "danger-button " : " "
+- warning_class = @color == "warning" ? "warning-button " : " "
+- disabled_class = @state == "disabled" ? "disabled-button " : " "
+.button-container
+    - if @type == "link"
+        %a.regular-button{:name => @name, onclick: "displayAnimation()", id: "button-animation-id", class: "#{@variant}-button "+danger_class+warning_class+disabled_class+slim_class, href: @href}
+            - if @icon != "" && @icon_type == "left"
+                = inline_svg_tag @icon, class: "#{@variant}-button-icon left-button-icon"
+            = @value
+            - if @icon != "" && @icon_type == "right"
+                = inline_svg_tag @icon, class: "#{@variant}-button-icon right-button-icon"
+    - else
+        %button.regular-button{:name => @name, onclick: "displayAnimation()", id: "button-animation-id", class: "#{@variant}-button "+danger_class+warning_class+disabled_class+slim_class, type: "#{@type == "submit" ? "submit" : "button"}"}
+            - if @icon != "" && @icon_type == "left"
+                = inline_svg_tag @icon, class: "#{@variant}-button-icon left-button-icon"
+            = @value
+            - if @icon != "" && @icon_type == "right"
+                = inline_svg_tag @icon, class: "#{@variant}-button-icon right-button-icon"
+
+- if @state == "animate"
+    .animation-container#loading-animation{class: danger_class+warning_class+slim_class}
+        .lds-ellipsis
+            %div
+            %div
+            %div
+            %div
+
+    :javascript
+        const button = document.getElementById("button-animation-id")
```

Comment:
this will not work when on the same page you have multiple buttons sharing the same id.